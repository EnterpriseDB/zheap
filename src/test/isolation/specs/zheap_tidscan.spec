# Scenarios that test sanity of zheap_get_latest_tid() function

setup
{
  CREATE TABLE tidscan (a int PRIMARY KEY, v varchar);
  INSERT INTO tidscan VALUES (1, NULL);
}

teardown
{
  DROP TABLE tidscan;
}

session "s1"
setup		{ BEGIN; }
step "s1u"	{ UPDATE tidscan SET a = a + 1; }
step "s1c"	{ COMMIT; }

session "s2"
setup		{ BEGIN; }
step "s2f"	{ DECLARE c CURSOR FOR SELECT a FROM tidscan; FETCH FIRST FROM c; }
step "s2u"	{ UPDATE tidscan SET a = a + 2, v = 'session2' WHERE CURRENT  OF c; }
step "s2c"	{ COMMIT; }
step "s2s"	{ SELECT * from tidscan ORDER BY a; }

permutation "s1u" "s2f" "s2u" "s1c" "s2c" "s2s"
