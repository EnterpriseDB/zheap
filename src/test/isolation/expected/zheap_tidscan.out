Parsed test spec with 2 sessions

starting permutation: s1u s2f s2u s1c s2c s2s
step s1u: UPDATE tidscan SET a = a + 1;
step s2f: DECLARE c CURSOR FOR SELECT a FROM tidscan; FETCH FIRST FROM c;
a              

1              
step s2u: UPDATE tidscan SET a = a + 2, v = 'session2' WHERE CURRENT  OF c; <waiting ...>
step s1c: COMMIT;
step s2u: <... completed>
step s2c: COMMIT;
step s2s: SELECT * from tidscan ORDER BY a;
a              v              

4              session2       
